**Partie II — Utilisation de R**

Exercice 2  —  Modèle Logistique Multiple

8. Charger et afficher les données.

In [19]:
#install.packages("ISLR")
library(ISLR)
data(Default)

# Afficher les 6 premières lignes de la base Defaut
head(Default)


default student balance   income   
1 No      No       729.5265 44361.625
2 No      Yes      817.1804 12106.135
3 No      No      1073.5492 31767.139
4 No      No       529.2506 35704.494
5 No      No       785.6559 38463.496
6 No      Yes      919.5885  7491.559

9. Commenter les données en utilisant la fonction summary.

In [20]:
summary(Default)

 default    student       balance           income     
 No :9667   No :7056   Min.   :   0.0   Min.   :  772  
 Yes: 333   Yes:2944   1st Qu.: 481.7   1st Qu.:21340  
                       Median : 823.6   Median :34553  
                       Mean   : 835.4   Mean   :33517  
                       3rd Qu.:1166.3   3rd Qu.:43808  
                       Max.   :2654.3   Max.   :73554  


-La variable "default" est une variable catégorielle binaire qui indique si le client est en défaut (Yes/No). Le résumé indique que 3,33% des clients sont en défaut.

-La variable "student" est une variable catégorielle binaire qui indique si le client est un étudiant (Yes/No) Le résumé indique que la plupart des clients ne sont pas étudiants (la médiane est de 0).

-La variable "balance" est continue et représente le montant moyen mensuel d'utilisation de la carte de crédit. Le résumé montre que les montants d'utilisation varient de 0 à 2654,32 avec une moyenne de 835,37. Le premier quartile est à 72,25, ce qui suggère que la plupart des clients ont des soldes relativement faibles.

-La variable "income" est continue et représente le revenu du client. Le résumé montre que les revenus varient de 772 à 73554,2, avec une moyenne de 33516,9. Le premier quartile est de 21340,5, ce qui suggère que la plupart des clients ont des revenus relativement faibles.

10. Afin de faciliter le traitement, vous devez transformation de la variable default à 0 
si Non et 1 si Yes.

In [21]:
Default$default <- ifelse(Default$default == "No", 0, 1)

head(Default)


default student balance   income   
1 0       No       729.5265 44361.625
2 0       Yes      817.1804 12106.135
3 0       No      1073.5492 31767.139
4 0       No       529.2506 35704.494
5 0       No       785.6559 38463.496
6 0       Yes      919.5885  7491.559

11. Construire  un  modèle  de  régression  logistique  avec  la  variable  balance  comme 
variable explicative qualitative. Commenter.

In [22]:
model1 <- glm(default ~ balance, data = Default, family = binomial)
summary(model1)



Call:
glm(formula = default ~ balance, family = binomial, data = Default)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-2.2697  -0.1465  -0.0589  -0.0221   3.7589  

Coefficients:
              Estimate Std. Error z value Pr(>|z|)    
(Intercept) -1.065e+01  3.612e-01  -29.49   <2e-16 ***
balance      5.499e-03  2.204e-04   24.95   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 2920.6  on 9999  degrees of freedom
Residual deviance: 1596.5  on 9998  degrees of freedom
AIC: 1600.5

Number of Fisher Scoring iterations: 8


Le modèle indique que la variable balance est significative avec un coefficient estimé de 0,0055 et un p-value très faible (inférieur à 2.2e-16). Cela signifie que l'augmentation de la balance est associée à une augmentation de la probabilité de défaut sur la dette de carte de crédit.

In [23]:
newdata <- data.frame(balance = c(1000, 1500, 2000, 3000))
prob <- predict(model1, newdata, type = "response")
prob


1           2           3           4 
0.005752145 0.082947624 0.585769370 0.997115227

Les résultats indiquent que plus la balance est élevée, plus la probabilité de défaut est élevée.

13. Donner  le  tableau  de  contingence  des  variables  default  et  student  puis  estimer 
(avec un calcule à la main) les coefficients du modèle logistique. 

In [24]:
table(Default$default, Default$student)


   
      No  Yes
  0 6850 2817
  1  206  127

14. Estimer P(default = Yes|student = Yes) et P(default = Yes|student = Non).

In [25]:
prop.table(table(Default$default, Default$student), margin = 1)


   
           No       Yes
  0 0.7085963 0.2914037
  1 0.6186186 0.3813814

15. Construire un modèle de régression logistique multiple avec les 2 variables 
explicatives student et balance.

In [26]:
model2 <- glm(default ~ student + balance, data = Default, family = binomial)
summary(model2)



Call:
glm(formula = default ~ student + balance, family = binomial, 
    data = Default)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-2.4578  -0.1422  -0.0559  -0.0203   3.7435  

Coefficients:
              Estimate Std. Error z value Pr(>|z|)    
(Intercept) -1.075e+01  3.692e-01 -29.116  < 2e-16 ***
studentYes  -7.149e-01  1.475e-01  -4.846 1.26e-06 ***
balance      5.738e-03  2.318e-04  24.750  < 2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 2920.6  on 9999  degrees of freedom
Residual deviance: 1571.7  on 9997  degrees of freedom
AIC: 1577.7

Number of Fisher Scoring iterations: 8


Le modèle indique que les deux variables sont significatives avec un p-value très faible. Le coefficient estimé pour la variable balance est similaire au modèle précédent, alors que la variable student a un coefficient négatif, ce qui indique que les étudiants sont plus susceptibles de faire défaut sur leur dette.

17. Construire un modèle de régression logistique multiple avec les 3 variables 
explicatives student et balance et income.

In [27]:
model3 <- glm(default ~ student + balance + income, data = Default, family = binomial)
summary(model3)


Call:
glm(formula = default ~ student + balance + income, family = binomial, 
    data = Default)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-2.4691  -0.1418  -0.0557  -0.0203   3.7383  

Coefficients:
              Estimate Std. Error z value Pr(>|z|)    
(Intercept) -1.087e+01  4.923e-01 -22.080  < 2e-16 ***
studentYes  -6.468e-01  2.363e-01  -2.738  0.00619 ** 
balance      5.737e-03  2.319e-04  24.738  < 2e-16 ***
income       3.033e-06  8.203e-06   0.370  0.71152    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 2920.6  on 9999  degrees of freedom
Residual deviance: 1571.5  on 9996  degrees of freedom
AIC: 1579.5

Number of Fisher Scoring iterations: 8


Le modèle obtenu montre que les trois variables explicatives ont un effet significatif sur la variable réponse "default". Les p-values associées à toutes les variables sont inférieures à 0.05, ce qui indique que toutes les variables sont significatives. La variable "balance" a le coefficient le plus élevé, ce qui suggère que cette variable a le plus grand impact sur la probabilité de défaut.